thoth-api/src/graphql/model.rs

```diff
@@ -2154,13 +2187,25 @@ impl MutationRoot {
         data: PatchContribution,
     ) -> FieldResult<Contribution> {
         context.token.jwt.as_ref().ok_or(ThothError::Unauthorised)?;
+        user_can_edit_work(data.work_id, context)?;
+
         let connection = context.db.get().unwrap();
 
-        use crate::schema::contribution::dsl::*;
-        let target = contribution
-            .filter(work_id.eq(&data.work_id))
-            .filter(contributor_id.eq(&data.contributor_id))
-            .filter(contribution_type.eq(&data.contribution_type));
+        use crate::schema::contribution::dsl;
+        let target = dsl::contribution
```

Comment:
I'm a bit lost with this one, why do we need to query contribution twice?

Reply:
Compare `delete_contribution`, which has similar logic (unchanged under this PR). `target` is just setting up a query but `contribution` is also executing it. There's some difference in the behaviour of `.find` and `.filter` which means we can't reuse the partial query `target` here in the same way as we can in, e.g., `update_publication` (or `delete_publication`). To be honest, I'm not clear on why, as the Diesel definitions suggest they should both return the same type - better ideas welcome!

Reply:
In  `delete_contribution` we get the result beforehand so we can return it upon deletion (otherwise `diesel::delete() returns nothing). But in this case I'm not sure why you'd want to get the record before, since it is returned upon update.

I guess what I don't understand is the need to check `user_can_edit_work(contribution.work_id, context)?;` twice. If we are querying the contribution using `.filter(dsl::work_id.eq(&data.work_id))` then why would we need to check `!(data.work_id == contribution.work_id)`?

Reply:
Ah, now I see! Yes, you're right, there's no need for that as it's not possible in practice for the user to change the `work_id`. The same problem exists in `update_issue`. I'll fix both.